In [64]:
# Use this cell to begin your analysis, and add as many as you would like!
library(readxl)
library(dplyr)
library(stringr)

# read in
prices <- read.csv('datasets/airbnb_price.csv')
rooms <- read_excel('datasets/airbnb_room_type.xlsx')
reviews <- read.csv('datasets/airbnb_last_review.tsv', sep = '\t')

In [65]:
# view first few rows of all dfs
head(prices)
head(rooms)
head(reviews)

listing_id price       nbhood_full               
1 2595       225 dollars Manhattan, Midtown        
2 3831       89 dollars  Brooklyn, Clinton Hill    
3 5099       200 dollars Manhattan, Murray Hill    
4 5178       79 dollars  Manhattan, Hell's Kitchen 
5 5238       150 dollars Manhattan, Chinatown      
6 5295       135 dollars Manhattan, Upper West Side

listing_id description                               room_type      
1 2595       Skylit Midtown Castle                     Entire home/apt
2 3831       Cozy Entire Floor of Brownstone           Entire home/apt
3 5099       Large Cozy 1 BR Apartment In Midtown East Entire home/apt
4 5178       Large Furnished Room Near B'way           private room   
5 5238       Cute & Cozy Lower East Side 1 bdrm        Entire home/apt
6 5295       Beautiful 1br on Upper West Side          Entire home/apt

listing_id host_name   last_review 
1 2595       Jennifer    May 21 2019 
2 3831       LisaRoxanne July 05 2019
3 5099       Chris       June 22 2019
4 5178       Shunichi    June 24 2019
5 5238       Ben         June 09 2019
6 5295       Lena        June 22 2019

In [66]:
# merge all dfs into one
df <- prices %>%
    left_join(rooms, by = 'listing_id') %>%
    left_join(reviews, by = 'listing_id')

head(df)

listing_id price       nbhood_full               
1 2595       225 dollars Manhattan, Midtown        
2 3831       89 dollars  Brooklyn, Clinton Hill    
3 5099       200 dollars Manhattan, Murray Hill    
4 5178       79 dollars  Manhattan, Hell's Kitchen 
5 5238       150 dollars Manhattan, Chinatown      
6 5295       135 dollars Manhattan, Upper West Side
  description                               room_type       host_name  
1 Skylit Midtown Castle                     Entire home/apt Jennifer   
2 Cozy Entire Floor of Brownstone           Entire home/apt LisaRoxanne
3 Large Cozy 1 BR Apartment In Midtown East Entire home/apt Chris      
4 Large Furnished Room Near B'way           private room    Shunichi   
5 Cute & Cozy Lower East Side 1 bdrm        Entire home/apt Ben        
6 Beautiful 1br on Upper West Side          Entire home/apt Lena       
  last_review 
1 May 21 2019 
2 July 05 2019
3 June 22 2019
4 June 24 2019
5 June 09 2019
6 June 22 2019

In [67]:
# find avg listing price
df <- df %>%
    mutate(price_clean = as.numeric(gsub(" dollars", "", df$price)))

avg_price <- mean(df$price_clean)

avg_price

[1] 141.7779

In [68]:
# count private rooms

unique(df$room_type)

nb_private_rooms <- df %>%
    mutate(room_type_clean = str_to_lower(room_type)) %>%
    filter(room_type_clean == 'private room') %>%
    count() %>%
    as.numeric()

nb_private_rooms

[1] "Entire home/apt" "private room"    "Private room"    "entire home/apt"
[5] "PRIVATE ROOM"    "shared room"     "ENTIRE HOME/APT" "Shared room"    
[9] "SHARED ROOM"

[1] 11356

In [69]:
# find earliest and most recent reviews

review_dates <- df %>%
    mutate(last_review_clean = as.Date(last_review, format = "%B %d %Y")) %>%
    summarize(first_reviewed = min(last_review_clean),
              last_reviewed = max(last_review_clean))

review_dates

first_reviewed last_reviewed
1 2019-01-01     2019-07-09

## Airbnb listings in New York City
<p><img src="https://assets.datacamp.com/production/project_1230/img/nyc.jpg" alt="New York City skyline" width="600px"></p>
<p>Welcome to New York City, one of the most-visited cities in the world. As a result, there are many <a href="https://www.airbnb.com/"><em>Airbnb</em></a> listings in New York City to meet the high demand for temporary lodging for anywhere between a few nights to many months. In this notebook, we will take a closer look at the New York Airbnb market by combining data from multiple file types like <code>.csv</code>, <code>.tsv</code>, and <code>.xlsx</code> (Excel files).</p>
<p>A <strong>CSV</strong>, or comma-separated-values, file is one of the most common ways that tabular data is stored. In a CSV file, each value is separated by a comma and each row is separated by a newline. Since this file format is so widely used and is non-proprietary, it's great for sharing data with others and can be parsed by a variety of software.</p>
<pre><code>"listing_id","price","nbhood_full"
2595,"225 dollars","Manhattan, Midtown"
3831,"89 dollars","Brooklyn, Clinton Hill"
5099,"200 dollars","Manhattan, Murray Hill"
</code></pre>
<p>A <strong>TSV</strong>, or tab-separated-values, file is similar to a CSV file, but tabs are used to separate values rather than commas:</p>
<pre><code>listing_id    host_name   last_review
2595    Jennifer    May 21 2019
3831    LisaRoxanne July 05 2019
5099    Chris   June 22 2019
</code></pre>
<p><strong>Excel files</strong> are often used by spreadsheet users. Excel files contain information about formatting and formulas created in Excel, but these things aren't usually necessary when working with data in R. Excel files can also contain multiple tables, so these files need to be imported carefully to make sure the correct table is used. The Excel file format is also proprietary, so there's a more limited pool of software that can read it.</p>
<p><img src="https://assets.datacamp.com/production/project_1230/img/airbnb_room_type.png" alt="Excel file" width="700px"></p>
<p>The three files that are available contain data on 2019 Airbnb listings. Here are the details:</p>
<div style="background-color: #efebe4; color: #05192d; text-align:left; vertical-align: middle; padding: 15px 25px 15px 25px; line-height: 1.6;">
<div style="font-size:20px"><b>datasets/airbnb_price.csv</b></div>
This is a CSV file containing data on the prices and neighborhoods of Airbnbs.
<ul>
<li><b><code>listing_id</code>:</b> unique identifier of listing</li>
<li><b><code>price</code>:</b> nightly listing price in USD</li>
<li><b><code>nbhood_full</code>:</b> name of borough and neighborhood where listing is located</li>
</ul>
</div>
<div style="background-color: #efebe4; color: #05192d; text-align:left; vertical-align: middle; padding: 15px 25px 15px 25px; line-height: 1.6;">
<div style="font-size:20px"><b>datasets/airbnb_room_type.xlsx</b></div>
This is an Excel file containing data on Airbnb listing descriptions and room types.
<ul>
<li><b><code>listing_id</code>:</b> unique identifier of listing</li>
<li><b><code>description</code>:</b> listing description</li>
<li><b><code>room_type</code>:</b> Airbnb has three types of rooms: shared rooms, private rooms, and entire homes/apartments</li>
</ul>
</div>
<div style="background-color: #efebe4; color: #05192d; text-align:left; vertical-align: middle; padding: 15px 25px 15px 25px; line-height: 1.6;">
<div style="font-size:20px"><b>datasets/airbnb_last_review.tsv</b></div>
This is a TSV file containing data on Airbnb host names and review dates.
<ul>
<li><b><code>listing_id</code>:</b> unique identifier of listing</li>
<li><b><code>host_name</code>:</b> name of listing host</li>
<li><b><code>last_review</code>:</b> date when the listing was last reviewed</li>
</ul>
</div>
<p><strong>Note:</strong> This project lets you apply the skills from the <a href="https://learn.datacamp.com/skill-tracks/importing-cleaning-data-with-r">Importing and Cleaning Data with R skill track</a>, including importing from different file types and cleaning numerical, categorical, and date data. We recommend that you take the courses in this track before starting this project.</p>